pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -532,8 +532,8 @@ func generateVirtualHostDomains(service *model.Service, port int, node *model.Pr
 //
 // - Given foo.local.campus.net on proxy domain "" or proxy domain example.com, this
 // function returns nil
-func GenerateAltVirtualHosts(hostname string, port int, proxyDomain string) []string {
-	if strings.Contains(proxyDomain, ".svc.") {
+func GenerateAltVirtualHosts(hostname string, port int, proxyDomain string, serviceIsLocal bool) []string {
+	if strings.Contains(proxyDomain, ".svc.") && serviceIsLocal {
```

Comment:
I think it should be based on the suffix matching with DNSDomain? Especially if this was working on 1.12 and older, which used DNSDomain

Reply:
Yeah, the resulting code when using DNSDomain suffix seems more explicit as to when hostnames are expanded or not.  It was easier to capture the nuance in comments at least!